src/modules/core/filter_pillar_echo.c

```diff
@@ -0,0 +1,314 @@
+/*
+ * filter_pillar_echo.c -- filter to interpolate pixels outside an area of interest
+ * Copyright (c) 2020-2021 Meltytech, LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include "image_proc.h"
+
+#include <framework/mlt.h>
+#include <framework/mlt_log.h>
+
+#include <math.h>
+#include <string.h>
+
+/** Constrain a rect to be within the max dimensions
+*/
+static mlt_rect constrain_rect( mlt_rect rect, int max_x, int max_y )
+{
+	if ( rect.x < 0 )
+	{
+		rect.w = rect.w + rect.x ;
+		rect.x = 0;
+	}
+	if ( rect.y < 0 )
+	{
+		rect.h = rect.h + rect.y;
+		rect.y = 0;
+	}
+	if ( rect.x + rect.w < 0 )
+	{
+		rect.w = 0;
+	}
+	if ( rect.y + rect.h < 0 )
+	{
+		rect.h = 0;
+	}
+	if ( rect.x + rect.w > max_x )
+	{
+		rect.w = max_x - rect.x;
+	}
+	if ( rect.y + rect.h > max_y )
+	{
+		rect.h = max_y - rect.y;
+	}
+	return rect;
+}
+
+typedef struct
+{
+	mlt_image src;
+	mlt_image dst;
+	mlt_rect rect;
+} scale_sliced_desc;
+
+static int scale_sliced_proc(int id, int index, int jobs, void* data)
+{
+	(void) id; // unused
+	scale_sliced_desc* desc = ((scale_sliced_desc*) data);
+	mlt_image src = desc->src;
+	mlt_image dst = desc->dst;
+	mlt_rect rect = desc->rect;
+	int slice_line_start, slice_height = mlt_slices_size_slice(jobs, index, src->height, &slice_line_start);
+	int slice_line_end = slice_line_start + slice_height;
+	double srcScale = rect.h / (double)src->height;
+	int linesize = src->width * 4;
+	uint8_t* d = dst->data + (slice_line_start * linesize);
+	int y = 0;
+	for ( y = slice_line_start; y < slice_line_end; y++ )
```

Comment:
Just a FYI for a while now we have supported using C99 so you can declare within a for loop, for example.

Reply:
Nice! Done.